### PR TITLE
.github/labeler.yml: add automatic labeling for more topics

### PR DESCRIPTION
**Pull request by jopejoe1**
We have those labels, no reason not to automatic assign them to prs that touch the specific areas.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -86,6 +141,12 @@
         - pkgs/build-support/flutter/*.nix
         - pkgs/development/compilers/flutter/**/*.nix
 
+"6.topic: games":
+  - any:
+    - changed-files:
+      - any-glob-to-any-file:
+        - pkgs/games/**/*
```
Comment (FliegendeWurst) [inline .github/labeler.yml:148]: Probably won't see much use once pkgs/by-name migration is complete.

File: .github/labeler.yml [hunk 2]
```diff
@@ -118,6 +185,25 @@
         - pkgs/top-level/haskell-packages.nix
         - pkgs/top-level/release-haskell.nix
 
+"6.topic: java":
+  - any:
+    - changed-files:
+      - any-glob-to-any-file:
+        - nixos/modules/programs/java.nix
+        - pkgs/development/compilers/openjdk/**/*
```
Comment (FliegendeWurst) [inline .github/labeler.yml:193]:
```suggestion
        - pkgs/development/compilers/graalvm/**/*
        - pkgs/development/compilers/openjdk/**/*
```

Comment (Scrumplex) [reply .github/labeler.yml:193]: also `pkgs/development/compilers/zulu` and `pkgs/development/compilers/temurin-bin`

**Review by Scrumplex — APPROVED** [review]

**Comment by zowoq** [comment]
darwin and bsd labels are managed by ofborg. See https://github.com/NixOS/nixpkgs/pull/348183, https://github.com/NixOS/nixpkgs/pull/164517

**Comment by jopejoe1** [comment]
> darwin and bsd labels are managed by ofborg. See #348183, #164517

Will open a pr to fix that shortly.

**Pull request merged**